### PR TITLE
Use gnu99 flag for tree-sitter and radare2-newshell files

### PR DESCRIPTION
**Pull request by ret2libc**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

If possible, we should use `-std=gnu99` on compilers that support it. This should not be relevant for all modern compilers, as they can easily compile tree-sitter and radare2-shell-parser code, but on old compilers (e.g. CentOS 6 / RHEL6) if we don't pass that flag it won't work because of some unnamed structs/unions used in the source files.

This PR checks for support of `-std=gnu99` and use that flag.

BTW, if there is not one already, it would be great if ACR would support CHECK_COMPILER_FLAG.
 
**Test plan**

CI should be green. I'm still working on merging https://github.com/tree-sitter/tree-sitter/pull/607 , then we can compile radare2 again on such an old system as CentOS 6.

### CODE REVIEW

**Comment by ret2libc** [comment]
By the way the tree-sitter PR was just merged, so I can probably include those changes here as well and with that we should be able to compile again on old compilers.

**Comment by radare** [comment]
I dont like adding flags that enable nonstandard features that depend on the compiler C is C. We dont need gnuisms

**Comment by ret2libc** [comment]
We are already using it for sdb, even on Mac. You can check it in our ci as well. New compilers won't have problems with that. Old compilers will use the feature only if supported, otherwise they won't. But some older compilers need that flag to compile some code. What's the real issue?

File: configure.acr [hunk 1]
```diff
@@ -84,6 +84,30 @@ IFNOT HAVE_CLOCK_GETTIME {
...
+echo "int main(int argc, char **argv) { return 0; }" > temp.c
+${CC} -std=gnu99 temp.c -o temp.c.bin
+SUPPORT_GNU99=$?
+rm -f temp.c temp.c.bin
+<<EOF
+
+IF SUPPORT_GNU99 {
+	NEWSHELL_CFLAGS += '-std=gnu99' ;
+}{
+	CHECK SUPPORT_C99 <<EOF
```
[the remark this answers is not in the document]
Comment (radare) [reply configure.acr:100]: wait for acr 1.9

[the remark this answers is not in the document]
Comment (radare) [reply configure.acr:100]: acr 1.9 is out

**Comment by ret2libc** [comment]
Closed in favour of https://github.com/radareorg/radare2/pull/16765

**Pull request closed without merging**